sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileSystemClient.cs

```diff
@@ -798,7 +972,167 @@ private void SetNameFieldsIfNull()
         public virtual async Task<Response<FileSystemInfo>> CreateIfNotExistsAsync(
             Models.PublicAccessType publicAccessType = Models.PublicAccessType.None,
             Metadata metadata = default,
+            DataLakeFileSystemEncryptionScopeOptions encryptionScopeOptions = default,
             CancellationToken cancellationToken = default)
+        {
+            DiagnosticScope scope = ClientConfiguration.ClientDiagnostics.CreateScope($"{nameof(DataLakeFileSystemClient)}.{nameof(CreateIfNotExists)}");
+            try
+            {
+                scope.Start();
+                Response<BlobContainerInfo> containerResponse = await _containerClient.CreateIfNotExistsAsync(
+                    publicAccessType: (Blobs.Models.PublicAccessType)publicAccessType,
+                    metadata: metadata,
+                    encryptionScopeOptions: encryptionScopeOptions.ToBlobContainerEncryptionScopeOptions(),
+                    cancellationToken: cancellationToken).ConfigureAwait(false);
```

Comment:
This seems to be repeated above, can we extract common logic into internal methods?

Reply:
Normally, we would be calling `CreateInternal()` here.  In this case, we want to take advantage of the [BlobContainerClient.CreateIfNotExists()](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs#L1041) logic for handling the different error codes.  Otherwise, we would basically have to re-implement the error code logic here.

Reply:
ah, so we can't use async flag here... nvm.